plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx

```diff
@@ -23,13 +23,14 @@ import {
   SearchModalProvider,
   useSearchModal,
 } from '../SearchModal';
+import { SearchResultChildrenProvider } from '../../plugin';
 
 /**
  * Props for {@link SidebarSearchModal}.
  *
  * @public
  */
-export type SidebarSearchModalProps = {
+export type SidebarSearchModalProps = SearchResultChildrenProvider & {
   icon?: IconComponent;
   children?: (props: SearchModalChildrenProps) => JSX.Element;
 };
```

Comment:
We can pick the `resultItem` type from the `SearchModalProps` 🙂 :

```suggestion
export type SidebarSearchModalProps = Pick<
  SearchModalProps,
  'children' | 'resultItems'
> & {
  icon?: IconComponent;
};
```

Reply:
But what if you want to provide both?

Reply:
In this case we are picking both children and resulItems, not sure if I understand the question

Reply:
Sorry, I misunderstood what you meant. I was still under the impression that you could provide children and resultItemComponents